lib/statsd.js

```diff
@@ -197,6 +197,13 @@ Client.prototype.sendAll = function (stat, value, type, sampleRate, tags, callba
     tags = undefined;
   }
 
+  if (value === undefined) {
+    if (typeof callback === 'function') {
+      callback(null, 0);
+    }
+    return;
+  }
```

Comment:
This feels like an error case, since nothing is being sent as expected.  I would be ok with this if it sends on an error message in the callback and called errorHandler if that was present, similar to the error code below.

I also would like to see a test for this.

If those things can be changed, I'm good with adding this.  I would want to let it sit in review for a little bit, just in case someone else knows of a case (with DogStatsD or Telegraf) where this would want to send undefined, although I don't know of any myself.

Reply:
I wanted it to behave the same as before (it just wouldn't send anything at all), but I agree that we should see what happens in the other services.

Reply:
I'm concerned however that it does not actually behave the same as before- without knowing how all endpoints react to this, we could be changing behavior.  It's also not sending when the function says it would, so I do want to see this go through the error case to merge this in.  Also I will need a test for the case.  Thanks.

Reply:
That's ok. We'll deal with this in our wrapper package. Thank you for your time.